### ISSUE TITLE
ExceptionInInitializerError caused by NumberFormatException in ReflectionHelper.getJavaVersion

### ISSUE THREAD

**Issue by marcosbozzani** [SvenEwald/xmlbeam#60]
It seems that in Android `java.specification.version` returns `0.9` (https://developer.android.com/reference/java/lang/System#getProperties()) and `ReflectionHelper.getJavaVersion` expects an Integer

https://github.com/SvenEwald/xmlbeam/blob/cae9215985fe9c82818701db187fd2f078a8f3d6/src/main/java/org/xmlbeam/util/intern/ReflectionHelper.java#L616

# Logcat

```
java.lang.ExceptionInInitializerError
        at org.xmlbeam.types.DefaultTypeConverter.<init>(DefaultTypeConverter.java:273)
        at org.xmlbeam.XBProjector.<init>(XBProjector.java:563)
        at org.xmlbeam.XBProjector.<init>(XBProjector.java:600)
        at com.exceptionsample.MainActivity.onCreate(MainActivity.java:15)
       ...
     Caused by: java.lang.NumberFormatException: For input string: "0.9"
        at java.lang.Integer.parseInt(Integer.java:615)
        at java.lang.Integer.parseInt(Integer.java:650)
        at org.xmlbeam.util.intern.ReflectionHelper.getJavaVersion(ReflectionHelper.java:616)
        at org.xmlbeam.util.intern.ReflectionHelper.<clinit>(ReflectionHelper.java:63)
        at org.xmlbeam.types.DefaultTypeConverter.<init>(DefaultTypeConverter.java:273) 
        at org.xmlbeam.XBProjector.<init>(XBProjector.java:563) 
        at org.xmlbeam.XBProjector.<init>(XBProjector.java:600) 
        at com.exceptionsample.MainActivity.onCreate(MainActivity.java:15) 
       ...
```

# Steps to reproduce

1. Create a project in Android Studio
2. Add `implementation 'org.xmlbeam:xmlprojector:1.4.20'` to `build.gradle`
3. Add `import org.xmlbeam.XBProjector;` to `MainActivity.java`
4. Add `XBProjector xbProjector = new XBProjector();` to `MainActivity.onCreate(Bundle savedInstanceState)`
5. Run the project

# Environment

```
Android Studio Arctic Fox | 2020.3.1 Patch 4
Build #AI-203.7717.56.2031.7935034, built on November 21, 2021
Runtime version: 11.0.10+0-b96-7249189 amd64
VM: OpenJDK 64-Bit Server VM by Oracle Corporation
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 2
Registry: ide.instant.shutdown=false, external.system.auto.import.disabled=true

```

**Comment by SvenEwald** [comment]
I'm not quite sure if I can promise support for the Android JVMs. My efforts to support Java 6 up to Java 17 are more and more depending on deeper insight in the JDK implementation details. This looks like a dead end, supporting one Version will break support for another sooner or later. However, I will try to fix this issue in 1.4.21.

**Comment by SvenEwald** [comment]
Verified error with 1.4.20 in Android Studio.
Good news: Verrified that 1.4.21 fixes at least this issue.
